### ISSUE TITLE
training/eval loss doesn't decrease when using paged_adamw_8bit

### ISSUE THREAD

**Issue by KKcorps** [artidoro/qlora#78]
`paged_adamw_32bit` works perfectly though. Have tried this on multiple models, all of them lead to same observation.

## With paged_adamw_8bit

```
{'loss': 2.183, 'learning_rate': 0.0002, 'epoch': 0.14}                                                   
 10%|██████▊                                                             | 10/100 [01:43<15:04, 10.05s/it]Saving PEFT checkpoint...

{'loss': 3.3851, 'learning_rate': 0.0002, 'epoch': 0.29}                                                  
 20%|█████████████▌                                                      | 20/100 [03:20<13:07,  9.85s/it]Saving PEFT checkpoint...

{'loss': 9.2042, 'learning_rate': 0.0002, 'epoch': 0.43}                                                  
 30%|████████████████████▍                                               | 30/100 [04:59<11:21,  9.73s/it]Saving PEFT checkpoint...

{'loss': 50.7927, 'learning_rate': 0.0002, 'epoch': 0.57}                                                 
 40%|███████████████████████████▏                                        | 40/100 [06:37<10:03, 10.05s/it]Saving PEFT checkpoint...

 43%|█████████████████████████████▏                                      | 43/100 [07:08<09:31, 10.02s/it]
```

## With paged_adamw_32bit
```
{'loss': 2.1039, 'learning_rate': 0.0002, 'epoch': 0.14}                                                  
 10%|██████▊                                                             | 10/100 [01:46<15:36, 10.41s/it]Saving PEFT checkpoint...

{'loss': 1.7098, 'learning_rate': 0.0002, 'epoch': 0.29}                                                  
 20%|█████████████▌                                                      | 20/100 [03:27<13:11,  9.90s/it]Saving PEFT checkpoint...

{'loss': 1.5759, 'learning_rate': 0.0002, 'epoch': 0.43}                                                  
 30%|████████████████████▍                                               | 30/100 [05:08<11:23,  9.76s/it]Saving PEFT checkpoint...

{'loss': 1.4209, 'learning_rate': 0.0002, 'epoch': 0.57}                                                  
 40%|███████████████████████████▏                                        | 40/100 [06:47<10:08, 10.14s/it]Saving PEFT checkpoint...

{'loss': 1.2813, 'learning_rate': 0.0002, 'epoch': 0.71}                                                  
 50%|██████████████████████████████████                                  | 50/100 [08:27<08:05,  9.71s/it
 ```

**Comment by BugReporterZ** [comment]
If you configure the learning rate to be in the order of 1000 times smaller than with the 32bit version, on the long term the training loss appears to behave as expected (decreasing), but on the short term it has a sawtooth-like appearance. I am not sure if it's working as intended.
